example/jss/covid/toy/TruncatedPoisson.bl

```diff
@@ -6,18 +6,18 @@ model TruncatedPoisson {
   
   /** Mean parameter \(\lambda\). \(\lambda > 0\) */
  param RealVar mean
- random List<Double> factor ?: {
- 	return new ArrayList(9) => [
- 		for (int i : 3..<11){
- 			add(exp(i * log(mean)))
- 		}
- 	]
- }
+  
   laws {
     logf(realization, mean) {
       if (mean <= 0) return NEGATIVE_INFINITY
       if (realization < 3 || realization > 11) return NEGATIVE_INFINITY
-     return -log(factor.sum())+realization * log(mean)
+      val factor = new ArrayList(9) => [
+ 		for (int i : 3..<11){
+ 			add(exp(i * log(mean)))
+ 		}
+ 	]
+      val sumF=-log(factor.sum())
```

Comment:
That's great! A few thing to fix: you need the factorial part in there, and to include 11 in the bound I think, which you can do with``(3 .. 11)`` and just a tip to avoid creating the array:

```
(3..11).map[ expression you want to sum over here, with e.g. exp(it) ].reduce[x,y|x+y]
```

Getting close!